src/util/misc/matrix.ts

```diff
@@ -0,0 +1,207 @@
+import { cos, degreesToRadians, sin } from "..";
+import { iMatrix, PiBy180 } from "../../constants";
+import { IPoint, Point } from "../../point.class";
+import { TDegree, TMat2D } from "../../typedefs";
+
+type TRotateMatrixArgs = {
+  angle?: TDegree;
+}
+
+type TTranslateMatrixArgs = {
+  translateX?: number;
+  translateY?: number;
+}
+
+type TScaleMatrixArgs = {
+  scaleX?: number;
+  scaleY?: number;
+  flipX?: boolean;
+  flipY?: boolean;
+  skewX?: TDegree;
+  skewY?: TDegree;
+}
+
+type TComposeMatrixArgs = TTranslateMatrixArgs & TRotateMatrixArgs & TScaleMatrixArgs;
+/**
+ * Apply transform t to point p
+ * @static
+ * @memberOf fabric.util
+ * @param  {Point | IPoint} p The point to transform
+ * @param  {Array} t The transform
+ * @param  {Boolean} [ignoreOffset] Indicates that the offset should not be applied
+ * @return {Point} The transformed point
+ */
+export const transformPoint = (p: Point | IPoint, t: TMat2D, ignoreOffset: boolean): Point => {
+  if (ignoreOffset) {
+    return new Point(
+      t[0] * p.x + t[2] * p.y,
+      t[1] * p.x + t[3] * p.y
+    );
+  }
+  return new Point(
+    t[0] * p.x + t[2] * p.y + t[4],
+    t[1] * p.x + t[3] * p.y + t[5]
+  );
+};
```

Comment:
Probably a point.tansform is a good idea too.

Reply:
yes!